src/text-input-autocomplete.directive.ts

```diff
@@ -158,17 +162,27 @@ export class TextInputAutocompleteDirective implements OnDestroy {
         ),
         triggerCharacterPosition: this.elm.nativeElement.selectionStart
       };
+      //Get DOM element from component
+      this.menuElem = (this.menu.component.hostView as EmbeddedViewRef<any>)
+        .rootNodes[0] as HTMLElement;
+
+      // Append Menu element to the body
+      document.body.appendChild(this.menuElem);
```

Comment:
This will break universal, so you can access the document by adding this in the constructor: 
```
@Inject(DOCUMENT) private document
```